pkgs/development/ocaml-modules/mlx/ocamlmerlin-mlx.nix

```
@@ -0,0 +1,37 @@
+{
+  lib,
+  mlx,
+  buildDunePackage,
+  ppxlib,
+  merlin-lib,
+  cppo,
+  csexp,
+  menhir,
+  odoc,
+}:
+buildDunePackage {
+  pname = "ocamlmerlin-mlx";
+
+  inherit (mlx) version src;
+
+  minimalOCamlVersion = "4.14";
+
+  buildInputs = [
+    ppxlib
+    merlin-lib
+    csexp
+    menhir
+    odoc
+  ];
+
+  nativeBuildInputs = [
+    cppo
+  ];
+
+  meta = {
+    description = "Merlin support for MLX OCaml dialect";
+    homepage = "https://github.com/ocaml-mlx/mlx";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.Denommus ];
+  };
```

```suggestion
    mainProgram = "ocamlmerlin-mlx";
  };
```

Would you set the `meta.mainProgram` property? [ref](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#:~:text=meta.mainProgram%20must%20be%20set%20to%20the%20name%20of%20the%20executable%20which%20facilitates%20the%20primary%20function%20or%20purpose%20of%20the%20package%2C%20if%20there%20is%20such%20an%20executable%20in%20%24bin/bin/%20(or%20%24out/bin/%2C%20if%20there%20is%20no%20%22bin%22%20output).)

---

@momeemt done